<a href="https://colab.research.google.com/github/je-blueink/Basics-for-statistics/blob/main/python_and_r/practice_ch10_R.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### 책 3. 다음은 어느 비누 생산공정에서 비누 부스러기 부산물의 양과 공정속도 자료이다. 생산공정은 1번, 2번으로 구분되어 있다.

**(1) 공정속도를 설명변수로 하고 부산물의 양을 반응변수로 한 회귀모형을 만들고, 해석하시오.**

In [ ]:
# 데이터 입력 및 확인
data1 <- read.csv('sample_data/soap.csv')
head(data1, 3)

생산공정 부산물의양 공정속도
1 1        218        100     
2 1        248        125     
3 1        360        220

In [ ]:
model <- lm(부산물의양~공정속도, data=data1)
summary(model)


Call:
lm(formula = 부산물의양 ~ 공정속도, data = data1)

Residuals:
    Min      1Q  Median      3Q     Max 
-58.389 -26.474  -3.038  33.555  47.611 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  64.0357    23.2488   2.754   0.0108 *  
공정속도      1.1963     0.1061  11.280 2.68e-11 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 34.3 on 25 degrees of freedom
Multiple R-squared:  0.8358,	Adjusted R-squared:  0.8292 
F-statistic: 127.2 on 1 and 25 DF,  p-value: 2.676e-11


적합된 회귀식 :
부산물의 양 = 64.0357 + 1.1963*공정속도

- 공정속도가 증가함에 따라 부산물의 양도 함께 증가한다.

F통계량의 p-값 : 2.676e-11 < 0.05 이므로 위 회귀모형은 통계적으로 의미가 있다.

결정계수 : 0.8353 으로, 위 회귀모형은 83.5%의 설명력을 갖는다.

**(2) 생산공정을 두 그룹으로 각각 구분하여, (1)번 회귀모형을 만들고, 해석하시오.**

In [ ]:
# 생산공정에 따라 두 그룹으로 나누기
soap1 <- data1[data1$생산공정==1, -1]
soap2 <- data1[data1$생산공정==0, -1]
head(soap1, 3); head(soap2, 3)

부산물의양 공정속도
1 218        100     
2 248        125     
3 360        220

부산물의양 공정속도
16 140        105     
17 277        215     
18 384        270

In [ ]:
# 1번 생산공정 그룹 회귀모형
model1 <- lm(부산물의양~공정속도, data=soap1)
summary(model1)


Call:
lm(formula = 부산물의양 ~ 공정속도, data = soap1)

Residuals:
    Min      1Q  Median      3Q     Max 
-34.500 -14.403   5.496  16.046  28.418 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 97.96533   20.51430   4.775 0.000362 ***
공정속도     1.14539    0.09578  11.959 2.18e-08 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 22.19 on 13 degrees of freedom
Multiple R-squared:  0.9167,	Adjusted R-squared:  0.9103 
F-statistic:   143 on 1 and 13 DF,  p-value: 2.177e-08


In [ ]:
# 2번 생산공정 그룹 회귀모형
model2 <- lm(부산물의양~공정속도, data=soap2)
summary(model2)


Call:
lm(formula = 부산물의양 ~ 공정속도, data = soap2)

Residuals:
   Min     1Q Median     3Q    Max 
-23.93 -10.60  -6.22  12.88  39.51 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  7.57446   18.81855   0.402    0.696    
공정속도     1.32205    0.08352  15.829 2.08e-08 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 18.71 on 10 degrees of freedom
Multiple R-squared:  0.9616,	Adjusted R-squared:  0.9578 
F-statistic: 250.6 on 1 and 10 DF,  p-value: 2.083e-08


#### 1번 생산공정
적합된 회귀식 : 부산물의 양 = 97.965 + 1.145*공정속도
- 공정속도가 증가함에 따라 부산물의 양도 함께 증가한다.

F통계량의 p-값 : 2.177e-08 < 0.05 이므로 위 회귀모형은 통계적으로 의미가 있다.

결정계수 : 0.9167 로, 위 회귀모형은 91.7%의 설명력을 갖는다.


#### 2번 생산공정
적합된 회귀식 : 부산물의 양 = 7.574 + 1.322*공정속도
- 공정속도가 증가함에 따라 부산물의 양도 함께 증가한다.

F통계량의 p-값 : 2.083e-08 < 0.05 이므로 위 회귀모형은 통계적으로 의미가 있다.

결정계수 : 0.9616 으로, 위 회귀모형은 96.1%의 설명력을 갖는다.

**(3) 공정속도 및 생산공정을 설명변수로 하고, 부산물의 양을 반응변수로 한 회귀모형을 만들고, 해석하시오. (2)번의 결과와 비교하여 보시오.**

In [ ]:
model3 <- lm(부산물의양 ~ ., data=data1)
summary(model3)


Call:
lm(formula = 부산물의양 ~ ., data = data1)

Residuals:
    Min      1Q  Median      3Q     Max 
-39.557 -14.161  -0.121  17.518  33.953 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 27.28179   15.40701   1.771   0.0893 .  
생산공정    53.12920    8.21003   6.471 1.08e-06 ***
공정속도     1.23074    0.06555  18.775 7.48e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 21.13 on 24 degrees of freedom
Multiple R-squared:  0.9402,	Adjusted R-squared:  0.9352 
F-statistic: 188.6 on 2 and 24 DF,  p-value: 2.104e-15


적합된 회귀식 : 부산물의 양 = 27.282 + 53.129생산공정 + 1.231공정속도
- 공정속도가 증가함에 따라 부산물의 양도 함께 증가한다.
- 생산공정과 부산물의 양은 양의 상관관계를 갖는다. 즉, 2번 생산공정(값=0)보다 1번 생산공정(값=1)에서 부산물의 양이 더 많다

F통계량의 p-값 : 2.104e-15 < 0.05 이므로 위 회귀모형은 통계적으로 의미가 있다.

모든 회귀계수의 p-값이 0.05 이하이고, 매우 작은 값이므로, 모든 회귀계수는 통계적으로 유의미하다.

수정된 결정계수 : 0.9352 로, 위 회귀모형은 93.5%의 설명력을 갖는다.

(2)번의 결과와 비교하면, 생산공정을 구분해서 각각 회귀모형을 만들었을 때보다  F통계량의 값이 훨씬 작아졌으므로 설명변수에 생산공정을 포함해서 만든 회귀모형이 통계적으로 더 유의미하다.
- model1 F통계량 2.177e-08 / model2 F통계량 2.083e-08 / model3 F통계량 2.104e-15



### 워크북 4. R에서 health 데이터 객체에서 변수 time을 반응변수로, 나머지 변수를 설명변수로 한 회귀모형을 적합하고자 한다.



```
health = health[ , -1]
fit = lm(time ~ ., data=health)
```